GoogleDataTransport/GDTCCTLibrary/GDTCCTUploadOperation.m

```diff
@@ -78,14 +65,24 @@ @interface GDTCCTUploadOperation () <NSURLSessionDelegate>
 
 @property(nonatomic, readonly) GDTCORTarget target;
 @property(nonatomic, readonly) GDTCORUploadConditions conditions;
+@property(nonatomic, readonly) NSURL *uploadURL;
+@property(nonatomic, nullable, readonly) NSString *APIKey;
+@property(nonatomic, readonly) id<GDTCCTUploadMetadataProvider> metadataProvider;
 
 @property(nonatomic, readwrite, getter=isExecuting) BOOL executing;
 @property(nonatomic, readwrite, getter=isFinished) BOOL finished;
+
+@property(nonatomic, copy) dispatch_block_t block;
+
 @end
 
 @implementation GDTCCTUploadOperation
 
-- (instancetype)initWithTarget:(GDTCORTarget)target conditions:(GDTCORUploadConditions)conditions {
+- (instancetype)initWithTarget:(GDTCORTarget)target
+                    conditions:(GDTCORUploadConditions)conditions
+                     uploadURL:(NSURL *)uploadURL
+                        APIKey:(nullable NSString *)APIKey
```

Comment:
maybe API key should part of the metadata provider? Not all backends need one and it'd reduce the arity of this initializer (which is always good).

Reply:
Good idea, will do.